File: backend/src/main/java/io/blog/devlog/domain/sitemap/controller/SitemapController.java
package io.blog.devlog.domain.sitemap.controller;

import io.blog.devlog.domain.sitemap.dto.PostUrlDto;
import io.blog.devlog.domain.sitemap.service.SitemapService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
@RequestMapping("/sitemap")
@Slf4j
public class SitemapController {

    private final SitemapService sitemapService;

    @PostMapping()
    void generateSitemap() {
        try {
            sitemapService.createAllPostToSubSitemap();
            log.info("Sitemap generated");
        } catch (Exception e) {
            log.error("Sitemap generation failed", e);
            throw e;
        }
    }

    @PostMapping("/post")
    void addPostSitemap(@RequestBody PostUrlDto postUrl) {
        try {
            sitemapService.addPostToSubSitemap(postUrl);
            log.info("Post sitemap added : " + postUrl.getUrl());
        } catch (Exception e) {
            log.error("Post sitemap add failed : " + postUrl.getUrl(), e);
            throw e;
        }
    }

    @DeleteMapping("/post")
    void deletePostSitemap(@RequestBody PostUrlDto postUrl) {
        try {
            sitemapService.deletePostFromSubSitemap(postUrl);
            log.info("Post sitemap deleted : " + postUrl.getUrl());
        } catch (Exception e) {
            log.error("Post sitemap delete failed : " + postUrl.getUrl(), e);
            throw e;
        }
    }
}

File: backend/sitemap-service/src/main/kotlin/io/blog/sitemap/controller/SitemapController.kt
package io.blog.sitemap.controller

import io.blog.sitemap.dto.PostUrlDto
import io.blog.sitemap.service.SitemapService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class SitemapController(private val sitemapService: SitemapService) {

    @PostMapping("/sitemap")
    fun generateSitemap() {
        try {
            val generateSitemap = sitemapService.createAllPostToSubSitemap()
            println("Sitemap generated: $generateSitemap")
        } catch (e: Exception) {
            println("Error generating sitemap: $e")
            throw e
        }
    }

    @PostMapping("/sitemap/post")
    fun addPostSitemap(@RequestBody postUrl: PostUrlDto) {
        try {
            sitemapService.addPostToSubSitemap(postUrl)
            println("Post added to sitemap: $postUrl")
        } catch (e: Exception) {
            println("Error adding post to sitemap: $e")
            throw e
        }
    }

    @DeleteMapping("/sitemap/post")
    fun deletePostSitemap(@RequestBody postUrl: PostUrlDto) {
        try {
            sitemapService.deletePostFromSubSitemap(postUrl)
            println("Post deleted to sitemap: $postUrl")
        } catch (e: Exception) {
            println("Error deleting post to sitemap: $e")
            throw e
        }
    }
}